tests/test_request_header_authentication.py

```diff
@@ -0,0 +1,31 @@
+from unittest import mock  # noqa
+
+import pytest
+from werkzeug.test import EnvironBuilder
+
+from notifications_utils.request_helper import CustomRequest, _check_proxy_header_secret
+
+
+@pytest.mark.parametrize('test_input,expected', [
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key', 'old_key']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key', '']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'My-New-Header', 'header_value': 'right_key', 'secrets': ['right_key', '']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['', 'right_key']}, (True, 'Key used: 2')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['', 'old_key', 'right_key']}, (True, 'Key used: 3')),  # noqa
+    ({'secrets': ['old_key', 'right_key']}, (False, 'Header missing')),
```

Comment:
i'd pull out this into its own test since it has less in common with the other tests

Reply:
Done